src/FulfillerMultiSelect.jsx

```diff
@@ -62,26 +64,41 @@ class FulfillerMultiSelect extends React.Component {
     refreshComponentData() {
         return Promise.all([
             !this.props.fulfillers
-                ?
-                this.fetchFulfillers(this.props.accessToken)
-                :
-                Promise.resolve(),
+                ? this.fetchFulfillers(this.props.accessToken)
+                : Promise.resolve(),
             this.getRecentFulfillerIds()
         ])
-            .then(() => {
-                if (this.state.recentFulfillerIds.length) {
-                    this.setState(
-                        {
-                            selectedFulfillerIds: this.props.autoSelectMostRecent ?
-                                this.state.recentFulfillerIds :
-                                null
-                        },
-                        () => {
-                            if (this.props.onChange && this.props.autoSelectMostRecent) {
-                                this.props.onChange(this.state.recentFulfillerIds.map(f => this.fulfillerMap[f]));
-                            }
-                        });
+            .then(results => {
+                if (!results[1]) {
+                    return;
                 }
+
+                const { recentFulfillerIds, recentFulfillerIdsMulti } = results[1];
+                let fulfillerIdsToSelect = [];
+                if (this.props.selectedFulfillerIds) {
+                    fulfillerIdsToSelect = this.props.selectedFulfillerIds
+                } else if (recentFulfillerIds.length > 0 && this.props.autoSelectMostRecent) {
+                    fulfillerIdsToSelect = this.props.multi
+                        ? recentFulfillerIdsMulti
+                        : recentFulfillerIds.slice(0, 1);
+                }
+
+                const selectedFulfillerIds = fulfillerIdsToSelect
+                    .map(fulfillerId => (
+                        this.fulfillerMap[fulfillerId] && this.fulfillerMap[fulfillerId].fulfillerId
+                    ))
+                    .filter(fulfillerId => fulfillerId);
+
+                this.setState({
+                    recentFulfillerIds,
+                    selectedFulfillerIds
+                },
+                () => {
+                    if (this.state.selectedFulfillerIds.length > 0) {
+                        const updateCustomizrSettings = !!this.props.selectedFulfillerIds;
+                        this.onSelectedFulfillerChanged(this.state.selectedFulfillerIds, updateCustomizrSettings);
```

Comment:
Here is a small change to the customizr update logic. I removed most of the conditions from `updateRecentFulfillerIds` and let the `updateCustomizrSettings` flag be passed from outside. As for example in this case, when the component is mounted, we only want to do the update when the selected fulfiller ids were passed via props. Otherwise there's no need.